code/flusion/retrospective-experiments/flusion-retrospective.R

```diff
@@ -0,0 +1,94 @@
+# Create a retrospective flusion ensemble for a given reference date and
+# combination of component models
+
+library(dplyr)
+library(lubridate)
+library(readr)
+
+library(hubEnsembles)
+library(hubData)
+
+library(here)
+setwd(here::here())
+
+# Parse arguments.  We're expecting a call like
+# Rscript --vanilla flusion-retrospective.R 2023-12-30 UMass-gbq_qr UMass-gbq_qr_no_level
+args <- commandArgs(trailingOnly = TRUE)
+ref_date <- as.Date(args[1])
+models <- args[-1]
+
+#' load forecasts
+#'
+#' @param hub_path string specifying path to hub directory structure
+#' @param models character vector of model names to load
+#'
+#' @return data frame with model outputs
+load_forecasts <- function(hub_path, models) {
+  hub_con <- connect_hub(hub_path)
+
+  forecasts <- hub_con |>
+    dplyr::filter(
+      reference_date == ref_date,
```

Comment:
Not sure how formal you want this code to be, but whatever R linter I have hooked up to VSCode shows a `no visible binding for global variable` error for variables that appear in these dplyr pipes.

I was able to get them to disappear by doing a `library(rlang)` at the top of the file and then prefixing the offending variables with `.data$`